core/types/gen_receipt_json.go

```diff
@@ -16,6 +16,7 @@ var _ = (*receiptMarshaling)(nil)
 // MarshalJSON marshals as JSON.
 func (r Receipt) MarshalJSON() ([]byte, error) {
 	type Receipt struct {
+		Type              uint8          `json:"type,omitempty"`
```

Comment:
Just a mental note here, this will disable Types with id `0`. Just something to be aware if we ever want to wrap legacy txs with a zero.